src/main/java/tech/sourced/siva/IndexReader.java

```diff
@@ -56,31 +66,40 @@ public final synchronized Index getCompleteIndex() throws SivaException {
     private Index readIndex(final BaseIndex index) throws SivaException {
         try {
             // go to the end of the file
-            this.sivaFile.seek(this.sivaFile.length());
+            long offset = this.sivaFile.length();
+            this.sivaFile.seek(offset);
             while (true) {
                 if (this.sivaFile.getFilePointer() == 0) {
                     break;
                 }
 
-                long block = this.sivaFile.getFilePointer();
+                long endOfBlock = this.sivaFile.getFilePointer();
 
-                this.sivaFile.seek(this.sivaFile.getFilePointer() - INDEX_FOOTER_SIZE);
-                IndexFooter indexFooter = this.readIndexFooter();
-                this.sivaFile.seek(this.sivaFile.getFilePointer() - INDEX_FOOTER_SIZE
-                        - indexFooter.getIndexSize());
+                long startOfFooterOffset = this.sivaFile.getFilePointer() - INDEX_FOOTER_SIZE;
+                this.sivaFile.seek(startOfFooterOffset);
+
+                IndexFooter indexFooter = this.readFooter();
+
+                long startOfIndexOffset = endOfBlock - INDEX_FOOTER_SIZE
+                        - indexFooter.getIndexSize();
+                this.sivaFile.seek(startOfIndexOffset);
 
                 this.readSignature();
-                this.readIndexVersion();
+                this.readVersion();
 
-                for (int i = 0; i < indexFooter.getEntryCount(); i++) {
-                    index.add(this.readEntry(indexFooter, block));
+                for (long i = 0; i < indexFooter.getEntryCount(); i++) {
+                    index.add(this.readEntry(indexFooter, endOfBlock));
                 }
 
                 index.endIndexBlock();
 
+                checkIndexCrc(indexFooter, startOfIndexOffset);
+
                 // go to the next index
-                this.sivaFile.seek(this.sivaFile.getFilePointer() - indexFooter.getBlockSize()
-                        + INDEX_FOOTER_SIZE);
+                offset = this.sivaFile.getFilePointer() - indexFooter.getBlockSize()
```

Comment:
Why are you reusing the offset variable? That doesn't add functionality to the actual code, and can be misleading.

Reply:
I changed this to use other variable and I renamed too the other offset variables to avoid long names removing for that the `Offset` tail. They are clear enough without it.